pkg/dashboard/resource/function.go

```diff
@@ -734,6 +731,22 @@ func (fr *functionResource) getCreationStateUpdatedTimeout(request *http.Request
 	return timeoutDuration
 }
 
+// updateFunctionCRDState updates the CRD state of the function
+func (fr *functionResource) updateFunctionCRDState(ctx context.Context, function platform.Function, state functionconfig.FunctionState) error {
+
+	// only the function name and namespace are needed when updating the CRD state
+	updateFunctionOptions := &platform.UpdateFunctionOptions{
+		FunctionMeta: &functionconfig.Meta{
+			Name:      function.GetConfig().Meta.Name,
+			Namespace: function.GetConfig().Meta.Namespace,
+		},
+	}
+	if err := fr.getPlatform().UpdateFunctionState(ctx, updateFunctionOptions, state); err != nil {
+		return errors.Wrap(err, "Failed to update function state")
+	}
+	return nil
+}
+
```

Comment:
not sure why needed. u have patch options to patch desired state which is essentially that. perhaps a better code reuse?